Linear retriever top level option for normalizer (#129693)
elastic/elasticsearch: docs/reference/elasticsearch/rest-apis/retrievers/linear-retriever.md

`normalizer` {applies_to}`stack: ga 9.1`
:   (Optional, String)

>     The normalizer to use when combining results.

Reviewer: We use the phrase `top level` quite a bit in this document, so we should define it here as well for clarity. 

```suggestion
    The top-level normalizer to use when combining results.
```

Author: made the changes.

    See [normalizers](#linear-retriever-normalizers) for supported values.
    Required when `query` is specified.
    
    When used with the [multi-field query format](../retrievers.md#multi-field-query-format) (`query` parameter), normalizes scores per [field grouping](../retrievers.md#multi-field-field-grouping).

>     When used with `retrievers`, serves as the default normalizer for any sub-retriever that doesn't specify its own normalizer. Per-retriever normalizers always take precedence over the top-level normalizer.

Reviewer: I found `When used with retrievers` to be a bit ambiguous.

```suggestion
    Otherwise serves as the default normalizer for any sub-retriever that doesn't specify its own normalizer. Per-retriever normalizers always take precedence over the top-level normalizer.
```

## Example

> This example of a hybrid search weights KNN results five times more heavily than BM25 results in the final ranking, with a top-level normalizer applied to all retrievers.

Reviewer: I thought we were going to move this to the Examples file?

Author: This example already existed, and we modified it to include top-level. We initially considered also adding a top-level example under retrievers-example and even implemented it. However, after review (including feedback from @ioana), we decided to remove it.

              "k": 10,
              "num_candidates": 100
            }
          },
          "weight": 5.0
        },
        {
          "retriever": {
            "standard": {
              "query": {
                "match": {
                  "title": "elasticsearch"
                }
              }
            }
          },
          "weight": 1.5
        }
      ]
    }
  }
}
```

### Using top-level normalizer

This example shows how to use a top-level normalizer that applies to all sub-retrievers:

```console
GET my_index/_search
{
  "retriever": {
    "linear": {
      "retrievers": [
        {
          "retriever": {
            "standard": {
              "query": {
                "match": {
                  "title": "elasticsearch"
                }
              }
            }
          },
          "weight": 1.0
        },
        {
          "retriever": {
            "knn": {
              "field": "title_vector",
              "query_vector": [0.1, 0.2, 0.3],
              "k": 10,
              "num_candidates": 100
            }
          },
          "weight": 2.0
        }
      ],
      "normalizer": "minmax"
    }
  }
}
```

> In this example, the `minmax` normalizer is applied to both the standard retriever and the kNN retriever. The top-level normalizer serves as a default that can be overridden by individual sub-retrievers. When using the multi-field query format, the top-level normalizer is applied to all generated inner retrievers.

Author: I am unsure if this has to be documented here or under retrievers-example.md file.
